File: 문자열/문자열 압축.py
def solution(s):        
    answer = 1000
        
    for i in range(1,len(s)+1):        
        prev = s[:i]
        # 현재 idx
        j = i
        # 현재 카운트
        k = 1
        # answerStr
        currentAnswer = ''
        while True:                
            
            if prev != s[j:j+i]:
                if k > 1:
                    currentAnswer += str(k) + prev
                else:
                    currentAnswer += prev     
                k = 1
            else:
                k += 1
                
            prev = s[j:j+i]                        
            
            # 범위가 str 넘어가면 
            if  j + i > len(s):      
                if k > 1:
                    currentAnswer += str(k) + prev
                else:
                    currentAnswer += s[j:]
                break
            else:
                j += i
        
        answer = min(len(currentAnswer), answer)
                    
    return answer

solution('aabbaccc')

File: javascript/문자열압축.js
let fs = require('fs');
let input = fs.readFileSync('javascript/test.txt').toString();

solution(input);

"use strict";
function solution(s) {
    
    let answer = s.length;
    const maxsplit = Math.floor(s.length / 2);


    for (i = 1; i <= maxsplit; i++){
        let value = "";

        let count = 0;
        let prev = "";
        for (j = 0; j <= s.length; j += i) {
            if (prev === s.slice(j, j + i)) {
                count += 1;
            }
            else {
                if (count > 1){
                    value += count.toString() + prev;
                }
                else{
                    value += prev;
                }
                count = 1;
                prev = s.slice(j, j + i);
            }
        }         
        value += s.slice(j-i);

        answer = Math.min(answer, value.length);
    }

    console.log(answer);

    return answer;
}